src/plugins/discover/public/application/main/state_management/discover_saved_search_container.ts

```diff
@@ -412,6 +407,12 @@ function getSavedSearchFieldForComparison(
     return savedSearch.breakdownField || ''; // ignore the difference between an empty string and undefined
   }
 
+  if (fieldName === 'viewMode') {
+    // By default, viewMode: undefined is equivalent to documents view
+    // So they should be treated as same
+    return savedSearch.viewMode ?? VIEW_MODE.DOCUMENT_LEVEL;
+  }
```

Comment:
Tested and this PR fixes the issue 👍. one ask, this is the actual fix of the problem, right? Could you add a unit test for it in `discover_saved_search_container.test.ts`? it would be great to increase coverage in this case 🙏

Reply:
@kertal The actual fix is in `src/plugins/discover/public/application/main/state_management/utils/build_state_subscribe.ts`. Here it's only a stylistic choice.

Reply:
Added more tests

Reply:
Ah, got it, thx a lot for adding more tests anyway 👍